ts/features/idpay/payment/xstate/machine.ts

```diff
@@ -67,10 +67,17 @@ const createIDPayPaymentMachine = () =>
             onDone: {
               target: "AUTHORIZATION_SUCCESS"
             },
-            onError: {
-              actions: "setFailure",
-              target: "AUTHORIZATION_FAILURE"
-            }
+            onError: [
+              {
+                actions: "setFailure",
+                cond: "isBlockingFailure",
+                target: "AUTHORIZATION_FAILURE"
+              },
+              {
+                actions: ["setFailure", "showErrorToast"],
```

Comment:
while toasts now correctly show in the auth phase, if the pre-auth catches a 429 then we get a GENERIC error; is it expected? I'm guessing it's not mapped design-wise but I'm pretty sure that we have the means to avoid the navigation in case of a pre-auth 429.
what do you think?

Reply:
This should not happen. Did you try it? If we get a 429 we return a `TOO_MANY_REQUESTS` failure, which shows an error toast.

Reply:
Yes it is expected. In the pre-authorization we don't have the need to show an error toast. The user can still retry from the previous screen. Different case is the authorization screen, where there is the need to stay in the same screen to allow user to retry the authorization since the transaction is binded to him.